mainapp/views.py

```diff
@@ -226,14 +226,17 @@ class Maintenance(TemplateView):
 
 def mapdata(request):
     district = request.GET.get("district", "all")
-    data = cache.get("mapdata:" + district)
+    page = int(request.GET.get("page", "1"))
+    data = cache.get("mapdata:" + district + ',page:' + str(page))
     if data:
         return JsonResponse(list(data) , safe=False)
     if district != "all":
-        data = Request.objects.exclude(latlng__exact="").filter(district=district).values()
+        data = Request.objects.exclude(latlng__exact="").filter(district=district)\
+                   .order_by('-id').values()[(page - 1) * 100: page * 100]
```

Comment:
You need to use the database to do the pagenation. This is loading everyting to memory(will kill the machine).
Instead of .values()[(page - 1) * 100: page * 100] use a query

Reply:
And you dont need the .exclude(latlng__exact="") cause we are no doing maps anymore.

Reply:
You are right, removed values to let django do the limit on query-set.